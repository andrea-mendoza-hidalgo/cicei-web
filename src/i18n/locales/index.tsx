import en from './translations.en.json';
import es from './translations.es.json';
import { SharedEn, SharedEs } from '../../shared/components/locale';
import { PagesEn, PagesEs } from '../../components/pages/locales';
import ProjectEn from '../../components/project/locale/project.en.json';
import ProjectEs from '../../components/project/locale/project.es.json';
import PublicationsEn from '../../components/publication/locale/publication.en.json';
import PublicationsEs from '../../components/publication/locale/publication.es.json';
import ResearchersEn from '../../components/researchers/locale/researchers.en.json';
import ResearchersEs from '../../components/researchers/locale/researchers.es.json';
import NewsEs from '../../components/news/locale/news.es.json';
import NewsEn from '../../components/news/locale/news.en.json';

en.shared= SharedEn;
es.shared= SharedEs;
en.pages = PagesEn;
es.pages = PagesEs;
en.project = ProjectEn;
es.project = ProjectEs;
en.publication = PublicationsEn;
es.publication = PublicationsEs;
en.researchers = ResearchersEn;
es.researchers = ResearchersEs;
en.news = NewsEn;
es.news = NewsEs;



export {en, es};