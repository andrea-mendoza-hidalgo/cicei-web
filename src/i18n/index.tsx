import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import { es, en } from './locales';

const options = {
  debug: false,

  resources: {
    'es-ES': {
      shared: es.shared,
      pages: es.pages,
      project: es.project,
      publication: es.publication,
      researchers: es.researchers,
      news: es.news
    },
    'en-US': {
      shared: en.shared,
      pages: en.pages,
      project: en.project,
      publication: en.publication,
      researchers: en.researchers,
      news: en.news
    }
  },

  fallbackLng: 'es-ES',

  ns: ['shared', 'pages', 'project', 'publication'],

  defaultNS: 'shared',

  react: {
    wait: true
  }
};

i18n.use(initReactI18next).use(LanguageDetector).init(options);

export default i18n;
