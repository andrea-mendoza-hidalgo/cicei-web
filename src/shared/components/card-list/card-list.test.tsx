import React from 'react';
import { mount } from 'enzyme';
import '../../../i18n';
import CardsList from './card-list';
import { MemoryRouter } from 'react-router-dom';

const fakeNews = [
  {
    _id: 'j34h5jg3h242kh1',
    date: '2010/01/01',
    resume:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam rhoncus varius velit, sed efficitur erat. Etiam finibus nunc metus, ac sodales diam molestie in. Nam commodo sollicitudin turpis nec egestas. Vestibulum vitae mauris nulla. Nullam ultricies ultricies dapibus. Suspendisse et lacinia turpis. Integer tincidunt molestie porta. Morbi sed arcu efficitur eros eleifend facilisis. Nunc sit amet augue in lorem dapibus fermentum. Integer at enim id sapien ultricies pharetra ac nec felis. Sed semper risus et vehicula vestibulum. Vestibulum leo turpis, luctus id interdum eget, consectetur ac diam. Donec condimentum libero sed ex eleifend dictum.Mauris ac cursus ex. Vestibulum eu ante risus. Curabitur sit amet ante dolor. Praesent cursus, diam ac imperdiet feugiat, enim tellus elementum enim, sed viverra elit enim at dui. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque in consequat purus. Donec odio massa, ullamcorper id tortor in, finibus mollis lacus. Maecenas a sem nisl.Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nulla fermentum tincidunt mi, non lobortis ligula tincidunt eget. Vivamus elit massa, commodo a tellus id, vulputate finibus sapien. Cras at justo nec odio tempus commodo sed a enim. Maecenas semper enim ut dui sodales, eget imperdiet ante consectetur. Maecenas eleifend dolor ut imperdiet vulputate. Fusce non euismod nisi, nec gravida est. Aenean ac purus ornare, dictum arcu non, porta augue. Mauris hendrerit, libero sagittis congue venenatis, tellus mauris rhoncus felis, id efficitur sapien lectus in ligula. Praesent condimentum ipsum a tortor egestas, id semper nulla sodales. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Proin ac dolor ut arcu lacinia imperdiet. Ut aliquet eget ex et porta. Donec dapibus augue eu elementum maximus.',
    title: 'Lorem ipsum dolor sit amet',
    userOwner: '5f0eaa11af6d10604f49f166'
  },
  {
    _id: 'j34h5jg3h242kh1',
    date: '2010/01/01',
    title: 'Lorem ipsum dolor sit amet',
    resume:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam rhoncus varius velit, sed efficitur erat. Etiam finibus nunc metus, ac sodales diam molestie in. Nam commodo sollicitudin turpis nec egestas. Vestibulum vitae mauris nulla. Nullam ultricies ultricies dapibus. Suspendisse et lacinia turpis. Integer tincidunt molestie porta. Morbi sed arcu efficitur eros eleifend facilisis. Nunc sit amet augue in lorem dapibus fermentum. Integer at enim id sapien ultricies pharetra ac nec felis. Sed semper risus et vehicula vestibulum. Vestibulum leo turpis, luctus id interdum eget, consectetur ac diam. Donec condimentum libero sed ex eleifend dictum.Mauris ac cursus ex. Vestibulum eu ante risus. Curabitur sit amet ante dolor. Praesent cursus, diam ac imperdiet feugiat, enim tellus elementum enim, sed viverra elit enim at dui. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque in consequat purus. Donec odio massa, ullamcorper id tortor in, finibus mollis lacus. Maecenas a sem nisl.Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nulla fermentum tincidunt mi, non lobortis ligula tincidunt eget. Vivamus elit massa, commodo a tellus id, vulputate finibus sapien. Cras at justo nec odio tempus commodo sed a enim. Maecenas semper enim ut dui sodales, eget imperdiet ante consectetur. Maecenas eleifend dolor ut imperdiet vulputate. Fusce non euismod nisi, nec gravida est. Aenean ac purus ornare, dictum arcu non, porta augue. Mauris hendrerit, libero sagittis congue venenatis, tellus mauris rhoncus felis, id efficitur sapien lectus in ligula. Praesent condimentum ipsum a tortor egestas, id semper nulla sodales. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Proin ac dolor ut arcu lacinia imperdiet. Ut aliquet eget ex et porta. Donec dapibus augue eu elementum maximus.',
    userOwner: '5f0eaa11af6d10604f49f166'
  },
  {
    _id: 'j34h5jg3h242kh1',
    date: '2010/01/01',
    title: 'Lorem ipsum dolor sit amet',
    resume:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam rhoncus varius velit, sed efficitur erat. Etiam finibus nunc metus, ac sodales diam molestie in. Nam commodo sollicitudin turpis nec egestas. Vestibulum vitae mauris nulla. Nullam ultricies ultricies dapibus. Suspendisse et lacinia turpis. Integer tincidunt molestie porta. Morbi sed arcu efficitur eros eleifend facilisis. Nunc sit amet augue in lorem dapibus fermentum. Integer at enim id sapien ultricies pharetra ac nec felis. Sed semper risus et vehicula vestibulum. Vestibulum leo turpis, luctus id interdum eget, consectetur ac diam. Donec condimentum libero sed ex eleifend dictum.Mauris ac cursus ex. Vestibulum eu ante risus. Curabitur sit amet ante dolor. Praesent cursus, diam ac imperdiet feugiat, enim tellus elementum enim, sed viverra elit enim at dui. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque in consequat purus. Donec odio massa, ullamcorper id tortor in, finibus mollis lacus. Maecenas a sem nisl.Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nulla fermentum tincidunt mi, non lobortis ligula tincidunt eget. Vivamus elit massa, commodo a tellus id, vulputate finibus sapien. Cras at justo nec odio tempus commodo sed a enim. Maecenas semper enim ut dui sodales, eget imperdiet ante consectetur. Maecenas eleifend dolor ut imperdiet vulputate. Fusce non euismod nisi, nec gravida est. Aenean ac purus ornare, dictum arcu non, porta augue. Mauris hendrerit, libero sagittis congue venenatis, tellus mauris rhoncus felis, id efficitur sapien lectus in ligula. Praesent condimentum ipsum a tortor egestas, id semper nulla sodales. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Proin ac dolor ut arcu lacinia imperdiet. Ut aliquet eget ex et porta. Donec dapibus augue eu elementum maximus.',
    userOwner: '5f0eaa11af6d10604f49f166'
  }
];
describe('Card List Render Test', () => {
  it('CardList renders correctly ', () => {
    let card = mount(
      <MemoryRouter>
        <CardsList data={fakeNews} path={'/test'} emptyMessage={'empty'}/>{' '}
      </MemoryRouter>
    );
    expect(card).toMatchSnapshot();
  });

  it('Empty CardList renders correctly ', () => {
    let card = mount(
      <MemoryRouter>
        <CardsList data={[]} path={'/test'} emptyMessage={'empty'}/>{' '}
      </MemoryRouter>
    );
    expect(card).toMatchSnapshot();
  });
});
