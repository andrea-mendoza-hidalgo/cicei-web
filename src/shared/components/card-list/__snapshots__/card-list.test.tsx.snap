// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card List Render Test CardList renders correctly  1`] = `ReactWrapper {}`;

exports[`Card List Render Test Empty CardList renders correctly  1`] = `ReactWrapper {}`;
